### ISSUE TITLE
Prevent login after registration

### ISSUE THREAD

**Issue by mohammad-ghanbari** [Tivix/django-rest-auth#314]
When I register a new user, the new user is automatically logged in. How can I prevent that?

**Comment by Viatrak** [comment]
If you use allauth and set email verification to mandatory, by default django-rest-auth won't return a token on registration.

Alternatively, modify your URL registration to point to your own version of the RegisterView. The current implementation returns a token in the create() function.

**Comment by RNogales94** [comment]
I have the same problem, I'd like to use the same create but without the return, is it mandatory to return the token?

**Comment by KennyLepping** [comment]
Was this ever solved? I am currently experiencing this issue.

**Comment by cjohnson318** [comment]
@KennyLepping This solution is for [dj-rest-auth](https://github.com/iMerica/dj-rest-auth), a fork of this project, but I hope it helps. My use case was that I wanted admin users to create other users, with lesser scope. This was an internal business app that would not have open-to-the-whole-internet registration. The workflow I had in mind was that an admin would create a new user, and then ask them to reset their password to login. The password reset process would act as an email verification.

TL;DR: I subclassed the dj-rest-auth `RegisterView`. I everything is the same, except for the `perform_create(self, serializer)` method, where I removed a call to allauth's function `complete_signup(request, user, email_verification, success_url, signal_kwargs=None)`.

```python
# views.py

from allauth.account import app_settings as allauth_settings

from django.conf import settings

from dj_rest_auth.app_settings import (
    JWTSerializer, TokenSerializer, create_token,
)
from dj_rest_auth.utils import jwt_encode

from dj_rest_auth.registration.views import RegisterView


class CustomRegisterView(RegisterView):

    def perform_create(self, serializer):
        user = serializer.save(self.request)
        if allauth_settings.EMAIL_VERIFICATION != \
                allauth_settings.EmailVerificationMethod.MANDATORY:
            if getattr(settings, 'REST_USE_JWT', False):
                self.access_token, self.refresh_token = jwt_encode(user)
            elif not getattr(settings, 'REST_SESSION_LOGIN', False):
                # Session authentication isn't active either, so this has to be
                #  token authentication
                create_token(self.token_model, user, serializer)
        return user
```

Later, I use this as,

```python
from .views import CustomRegisterView
from .serializers import AdminCreateSerializer

class AdminView(CustomRegisterView):
    serializer_class = AdminCreateSerializer
```

Hope that helps.

**Comment by Mendez83** [comment]
@cjohnson318 I am also using dj-rest-auth with your same aproach where the user have to change their password via reset link.

I have created a custom serializer for the CustomRegisterView
`
class RegisterWithMailSendSerializer(RegisterSerializer):

    def save(self, request, **kwargs):
        adapter = get_adapter()
        user = adapter.new_user(request)
        self.cleaned_data = self.get_cleaned_data()
        # user = adapter.save_user(request, user, self, commit=False)
        user = adapter.save_user(request, user, self, commit=True)
        if "password1" in self.cleaned_data:
            try:
                adapter.clean_password(self.cleaned_data['password1'],
                                       user=user)
            except DjangoValidationError as exc:
                raise serializers.ValidationError(
                    detail=serializers.as_serializer_error(exc))
        user.save()
        self.custom_signup(request, user)
        setup_user_email(request, user, [])
        # pg = PasswordResetTokenGenerator()
        # pg_token = pg.make_token(user)
        frontend_site = settings.FRONTEND_APP_BASE_URL
        token_generator = kwargs.get('token_generator',
                                     default_token_generator)
        temp_key = token_generator.make_token(user)
        path = reverse(
            'password_reset_confirm',
            args=[user_pk_to_url_str(user), temp_key],
        )
        full_url = frontend_site + path
        context = {
            'current_site': frontend_site,
            'user': user,
            'password_reset_url': full_url,
            'request': request,
        }
        if app_settings.AUTHENTICATION_METHOD != app_settings.AuthenticationMethod.EMAIL:
            context['username'] = user_username(user)

        email = self.get_cleaned_data()['email']
        get_adapter(request).send_mail('password_reset_key', email, context)
        return user


    def custom_signup(self, request, user):
        print('CUSTOM SIGNUP FUNCTION')
`
and this send me a querystring containing uid and token for password reset BUT if I try to reset password with those value I get "invalid token".

How and where you manage the sending links to user for password reset?
